src/index.js

```diff
@@ -154,23 +156,32 @@ function performCheck(input: HTMLInputElement, body: FormData, url: string): Pro
   requests.set(input, xhr)
 
   xhr.open('POST', url, true)
-  xhr.setRequestHeader('Accept', 'text/html; fragment')
+  xhr.setRequestHeader('Accept', 'application/json, text/plain;q=0.9')
   const result = send(xhr, body)
   result.then(clear, clear)
   return result
 }
 
-function send(xhr: XMLHttpRequest, body: FormData): Promise<string> {
+function send(xhr: XMLHttpRequest, body: FormData): Promise<string | {text: string}> {
   return new Promise((resolve, reject) => {
     xhr.onload = function() {
+      let data = xhr.responseText
+      if (xhr.getResponseHeader('Content-Type') === 'application/json') {
+        data = JSON.parse(data)
```

Comment:
In the success path we really don't care about what the response text is, so I think this parsing should go into the catch block.

Reply:
So the reason why I do the parsing here is so that the user of the element doesn't have to check if the message is JSON in their app and parse conditionally.

I'm not sure how this could go into the catch block.